### PR TITLE
feat: add ImageGridSlice node

### PR DESCRIPTION
**Pull request by shrimbly**
## Summary

Adds an `ImageGridSlice` node (V3 IO schema) that splits an image into a rows × columns grid and returns the tiles as a single batched `IMAGE` output.

## Details

- Category `image/transform`; inputs `image`, `rows`, `columns` (each 1–16).
- Tiles are cut row-major with uniform floor-sized cells (`height // rows`, `width // columns`) so they stack into one tensor via `torch.cat`. Any remainder pixels at the right/bottom edge are dropped to keep tiles uniform and batchable.
- Output is a single `IMAGE` batch of `rows * columns` tiles.
- Registered in `ImagesExtension`.

## Review Focus

- The uniform floor-sizing trades a few edge pixels for a batchable output. If keeping every pixel is preferred, the alternative is variable-size tiles (not batchable) or padding — happy to adjust.

## Testing

- Smoke-tested the crop/batch logic: 3×3 on 30×30 → `(9, 10, 10, 3)`; 16×16 on 512×512 → `(256, 32, 32, 3)`; non-divisible 10×10 @ 3×3 → `(9, 3, 3, 3)`; rows clamp 99 → 16.

## Related

- Frontend UI (preview overlay + widget): Comfy-Org/ComfyUI_frontend#12656

### CODE REVIEW

File: comfy_extras/nodes_images.py [hunk 1]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
...
 
+class ImageGridSlice(IO.ComfyNode):
+    MAX_ROWS = 16
+    MAX_COLUMNS = 16
+
+    @classmethod
+    def define_schema(cls):
+        return IO.Schema(
+            node_id="ImageGridSlice",
+            display_name="Image Grid Slice",
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1251]: I would propose this, we're trying to go with verb first for nodes:


```suggestion
            display_name="Split Image Grid",
```

File: comfy_extras/nodes_images.py [hunk 2]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
...
+            category="image/transform",
+            search_aliases=["grid", "slice image", "split image", "crop grid"],
+            description="Slices an image into a grid of rows x columns tiles, returned as a single batch (row-major).",
+            inputs=[
+                IO.Image.Input("image"),
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1256]: Plural form for consistency, note this  impacts your function below so need to update every reference in the code:

```suggestion
                IO.Image.Input("images"),
```

File: comfy_extras/nodes_images.py [hunk 3]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
...
+                IO.Int.Input("rows", default=2, min=1, max=cls.MAX_ROWS),
+                IO.Int.Input("columns", default=2, min=1, max=cls.MAX_COLUMNS),
+            ],
+            outputs=[
+                IO.Image.Output(display_name="images"),
+            ],
+        )
+
+    @classmethod
+    def execute(cls, image, rows, columns) -> IO.NodeOutput:
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1266]:
```suggestion
    def execute(cls, images, rows, columns) -> IO.NodeOutput:
```

File: comfy_extras/nodes_images.py [hunk 4]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
...
+        rows = max(1, min(rows, cls.MAX_ROWS))
+        columns = max(1, min(columns, cls.MAX_COLUMNS))
+
+        _, height, width, _ = image.shape
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1270]:
```suggestion
        _, height, width, _ = images.shape
```

File: comfy_extras/nodes_images.py [hunk 5]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
...
+        tile_height = height // rows
+        tile_width = width // columns
+
+        tiles = []
+        for row in range(rows):
+            y_start = row * tile_height
+            for col in range(columns):
+                x_start = col * tile_width
+                tiles.append(
+                    image[
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1280]:
```suggestion
                    images[
```

**Review by alexisrolland — CHANGES_REQUESTED** [review]

File: comfy_extras/nodes_images.py [hunk 6]
```diff
@@ -1240,6 +1240,54 @@ def execute(cls, images, filename_prefix: str, format: dict) -> IO.NodeOutput:
         return IO.NodeOutput(ui={"images": results})
 
 
+class ImageGridSlice(IO.ComfyNode):
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1243]: Maybe we could align the class name with the title (please check whether it's unique):

```suggestion
class SplitImageGrid(IO.ComfyNode):
```

File: comfy_extras/nodes_images.py [hunk 7]
```diff
@@ -1262,6 +1310,7 @@ async def get_node_list(self) -> list[type[IO.ComfyNode]]:
             ImageScaleToMaxDimension,
             SplitImageToTileList,
             ImageMergeTileList,
+            ImageGridSlice,
```
Comment (alexisrolland) [inline comfy_extras/nodes_images.py:1313]: Update reference:

```suggestion
            SplitImageGrid,
```

**Comment by alexisrolland** [comment]
@shrimbly would you like to move forward with this PR? I'd be in favor of it if you can do the small tweaks discussed above. Not urgent, just following up...
